test/integration/test_push_docker_image.py

```diff
@@ -24,32 +24,50 @@ def normalize_request_name(name: str):
     return re.sub(r"^_+|_+$", "", name)
 
 
+@dataclass
+class DockerImageSpec:
+    repository: str
+    tag: str
+
+    @property
+    def name(self) -> str:
+        return f"{self.repository}:{self.tag}"
+
+
+@pytest.fixture(scope="session")
+def registry_image():
+    spec = DockerImageSpec("registry", "2")
+    client = docker.from_env()
+    if not client.images.list(spec.name):
+        _logger.debug("Pulling Docker image with Docker registry")
+        client.images.pull(spec.repository, spec.tag)
+    return spec
+
+
+@pytest.fixture(scope="session")
+def sample_docker_image(registry_image):
+    return registry_image
+
+
 @contextlib.contextmanager
-def tagged_image(
-        dss_docker_image: DssDockerImage,
-        registry: LocalDockerRegistry,
-        tag: str = None,
-):
+def tagged_image(old: DockerImageSpec, new: DockerImageSpec):
     """
     Prepend host and port of LocalDockerRegistry to the repository name of the
     DssDockerImage to enable pushing the image to the local registry.
     """
-    new_repo = f"{registry.host_and_port}/{dss_docker_image.repository}"
     client = docker.from_env()
-    tag = tag or dss_docker_image.version
-    _logger.info(f'tagging image to {new_repo}:{tag}')
-    image = client.images.get(dss_docker_image.image_name)
-    image.tag(new_repo, tag)
-    tagged = DssDockerImage(new_repo, tag)
-    tagged.registry = registry
+    _logger.debug(f'tagging image from {old.name} to {new.name}')
+    image = client.images.get(old.name)
+    image.tag(new.repository, new.tag)
+    tagged = DssDockerImage(new.repository, new.tag)
     try:
         yield tagged
     finally:
-        docker.from_env().images.remove(tagged.image_name)
+        docker.from_env().images.remove(new.name)
 
 
 @pytest.fixture(scope="session")
-def docker_registry(request):
+def docker_registry(request, registry_image):
     """
     Provide a context for creating a LocalDockerRegistry accepting
     parameter ``repository``.
```

Comment:
this description seems to be also a little bit outdated

Reply:
Yes, correct, thanks!
See next push.